spec/helpers/match_helper.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'digest'
+
+RSpec::Matchers.define(:be_same_file_as) do |expected_file_path|
+  match do |actual_file_path|
+    expect(md5_hash(actual_file_path)).to eq(md5_hash(expected_file_path))
+  end
+
+  def md5_hash(file_path)
+    Digest::MD5.hexdigest(File.read(file_path))
+  end
+end
```

Comment:
Love this.  You can define another matcher that runs the bake script and cleans up the "actual_output" so it isn't committed to git.

```ruby
RSpec::Matchers.define(:bake_correctly) do |book|
  match do |book|
    actual_file = Tempfile.new(book)
    `./bake -b #{book} -i spec/books/#{book}/input.xhtml -o #{actual_file.path}`
    expect("spec/books/#{book}/expected_output.xhtml").to be_same_file_as(actual_file.path)
  end
end
```

```ruby
# spec/main_spec.rb
...
  it 'bakes chemistry' do
    expect('chemistry').to bake_correctly
  end

  it 'bakes dummy' do
    expect('dummy').to bake_correctly
  end
...
```